asv_bench/benchmarks/merge.py

```diff
@@ -13,6 +15,45 @@ def setup(self, existing_elements):
             d[f"var{i}"] = i
         self.dataset = xr.merge([d])
 
-    def time_variable_insertion(self, existin_elements):
+        d = {"set_2_{i}": i for i in range(existing_elements)}
+        self.dataset2 = xr.merge([d])
+
+    def time_variable_insertion(self, existing_elements):
         dataset = self.dataset
         dataset["new_var"] = 0
+
+    def time_merge_two_datasets(self, existing_elements):
+        xr.merge([self.dataset, self.dataset2])
+
+    def time_dict_of_dataarrays_to_dataset(self, existing_elements):
+        # The idea here is to time how long it takes to go from numpy
+        # and python data types, to a full dataset
+        # See discussion
+        # https://github.com/pydata/xarray/issues/7224#issuecomment-1292216344
+        value = np.array(["0", "b"], dtype=str)
+        time = np.array([0, 1])
+        coords = dict(time=time)
+        data_vars = {
+            "long_variable_name_{i}": xr.DataArray(data=value, coords=coords)
```

Comment:
Is it intentional that the DataArray creation is still inside the test and not in the setup?

Also, adding the coords in every DataArray is a bit "unfair".
But it is a good test on its own :)

Reply:
Thank you for the quick review.

To be honest, my code looks like this. the reason is because of the stark warnings
https://docs.xarray.dev/en/stable/internals/variable-objects.html?highlight=Variable

> Variable is public API, but because of its incomplete support for labeled data, it is mostly intended for advanced uses, such as in xarray itself or for writing new backends.

Therefore, I was always creating "full DataArrays". before merging them.